src/sentry/stacktraces/functions.py

```diff
@@ -271,6 +273,32 @@ def get_function_name_for_frame(frame, platform=None):
         return trim_function_name(rv, frame.get("platform") or platform)
 
 
+def get_source_link_for_frame(frame: Frame) -> str | None:
+    """If source_link points to a GitHub raw content link, process it so that
+    we can return the GitHub equivalent with the line number, and use it as a
+    stacktrace link. Otherwise, return the link as is.
+    """
+    try:
+        source_link = frame.get("source_link")
+    except KeyError:
+        return None
+    parse_result = urlparse(source_link)
```

Comment:
we should catch exceptions from invalid urls

Reply:
does urlparse itself not throw an exception on invalid urls?

Reply:
From https://docs.python.org/3/library/urllib.parse.html
> The [urlsplit()](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlsplit) and [urlparse()](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) APIs do not perform validation of inputs. They may not raise errors on inputs that other applications consider invalid. They may also succeed on some inputs that might not be considered URLs elsewhere. Their purpose is for practical functionality rather than purity.

I tried with a gibberish URL and it did not throw anything

Reply:
hmm okay then you might be able to ignore me

Reply:
i think it's worthwhile keeping the validator since the intent is to use the generated github link in the ui as the stacktrace link; if the raw github user content is invalid to begin with, then the ui will link to a fake github link right?